scarlet/observation.py

```diff
@@ -231,3 +253,146 @@ def get_scene(self, scene):
             msg = "get_scene is currently only supported when the observation frame matches the scene"
             raise NotImplementedError(msg)
         return self.images
+
+
+
+class Combination(Scene):
+
+
+    def __init__(self, images, psfs=None, weights=None, wcs = None, filtercurve=None, padding=3, target_psf = None, structure = None):
+        super().__init__(images.shape, wcs=wcs, psfs=psfs, filtercurve=filtercurve)
+
+        self.images = np.array(images)
+        self.padding = padding
+
+        if weights is not None:
+            self._weights = np.array(weights)
+        else:
+            self._weights = 1
+
+        if target_psf is not None:
+            self.target_psf = target_psf
+        else:
+            self.target_psf = target_psf
+        self.structure = structure
+
+
+    def match(self, scene):
+
+        if self.wcs == None:
+            raise TypeError('WCS is actually mandatory, please provide one (tbdiscussed)')
+        if self.psfs is not None:
+            #Get pixel coordinates in each frame.
+            mask, over_lr, over_hr  = resampling.match_patches(scene.shape, self.shape,scene.wcs, self.wcs)
+            self._over_lr = over_lr
+            self._over_hr = over_hr
+            self._mask = mask
+
+
+            #Compute diff kernel at hr
+
+            whr = scene.wcs
+            wlr = self.wcs
+
+            #Reference PSF
+            if self.target_psf is None:
+
+                if np.size(np.shape(scene.psfs)) ==2:
+                    _target = scene.psfs
+                    _shape = scene.shape
+                elif np.size(np.shape(scene.psfs)) ==3:
+                    _target = scene.psfs[0,:,:]
+                    _shape = scene.shape[1:]
+                else:
+                    raise ValueError('Wrong dimensions for psfs. psfs should be of dimensions n1xn2 or nbxn1xn2')
+            else:
+                _target = self.target_psf
+            # 3) compute obs.psf in the frame of scene
+            resconv_op = []
+
+            for _psf in self.psfs:
+
+
+                #Computes spatially matching observation and target psfs. The observation psf is also resampled to the scene's resolution
+
+                new_target, observed_psf = resampling.match_psfs(_target, _psf, whr, wlr)
```

Comment:
I'm also not following what's going on here

Reply:
Well, that function also matches the physical support of both psfs. I kept the target psf inside this loop just in case bands might have psfs living in different supports. I am not sure the case will arise much, but everything the function does with that regard is to take high resolution psf and trim it to the common support the case may be. Any way, I am pretty sure we will have to review this at some point if we deal differently between SED in various resolutions.